### ISSUE TITLE
Change Preview Tab to include Widget with Code View

### ISSUE THREAD

**Issue by chimon2000** [erickzanardo/dashbook#33]
We have actually started using this and feel like it's a nice feature to bake into dashbook.

Poached from [flutter_catelog](https://github.com/X-Wei/flutter_catalog/).  It allows developers to view a preview of the widget but then also see the source code for that widget.

[Here](https://github.com/X-Wei/widget_with_codeview) is the library that we are currently using in tandem with dashbook.

**Comment by erickzanardo** [comment]
Dashbook kind of support something similar to that already, you can provide a codelink property when adding the chapter and it will create an icon to top of the page, you can check it working here: https://flame-engine.github.io/flame_example/#/

I think the only difference from what you are suggestion is that it would should the code directly on the page? Maybe on an iframe or webview?

**Comment by chimon2000** [comment]
Yup,  that's the only major difference, with the ability to copy the code directly.  I  can put together a working example if needed.

**Comment by emersonsiega** [comment]
I've created a custom decorator with the package `widget_with_codeview`.

It's very simple.. Just need to add an IconButton to show/hide the code...

**Comment by erickzanardo** [comment]
wow, that is such a simple solution! I will try adding something similar on Dashbook

**Comment by chimon2000** [comment]
@erickzanardo wondering if this is something you're still looking to add.

**Comment by erickzanardo** [comment]
Hey @chimon2000 ! Yes, definitely, but I am failing to find time to work on that, would you have interest in trying to submit a PR for this? I could try guiding you in case you need assistance.
